docs/src/design/changing_the_primal.md

```diff
@@ -0,0 +1,516 @@
+# Design Notes: Why can you change the primal computation?
+
+These design notes are to help you understand why ChainRules why [`rrule`](@ref).
+It explains why we have a function, that includes the computation of the primal result, and that returns the pullback as a closure.
+This structure is perhaps surprising to some AD authors, who might expect just a function that performs the pullback.
+It is particularly notable that you are able to change the primal computation when defining the `rrule`.
+We will illustrate in this document why this is a crucial capacity.
+
+
+!!! note what about `frule`?
+    Discussion here is focused on on reverse mode and `rrule`.
+    Similar concerns to apply to to forward mode and `frule`.
+    In forward mode these concerns lead to the fusing of the `pushforward` into `frule`.
+    All the examples given here also apply in forwards mode.
+    In fact in forwards mode there are even more opportunities to take advantage of sharing work between the primal and derivative computations.
+    A particularly notable example is in efficiently calculating the pushforward of solving a differential equation via expanding the system of equations to also include the derivatives before solving it.
+
+
+
+
+
+## The Journey to `rrule`
+
+Let's imagine a different system for rules, one that doesn't let you do this.
+This system is what a lot of AD systems have.
+It is what [Nabla.jl](https://github.com/invenia/Nabla.jl/)[^1] had originally.
```

Comment:
footnotes at end of sentences to avoid breaking flow when reading
```suggestion
It is what [Nabla.jl](https://github.com/invenia/Nabla.jl/) had originally.[^1]
```